src/components/CloudinaryContext/CloudinaryContext.js

```diff
@@ -17,41 +37,14 @@ import {CloudinaryContextType} from './CloudinaryContextType';
 class CloudinaryContext extends CloudinaryComponent {
   constructor(props, context) {
     super(props, context);
-
-    this.calcState = this.calcState.bind(this);
-    this.state = this.calcState();
-  }
-
-  /**
-   * Calculates current state
-   * @returns {context, childrenProps, includeOwnBody}
-   */
-  calcState() {
-    const context = this.context || {}; //context might not exist
-    const props = {...context, ...this.props}; //merge context with props
-
-    //split the props to cloudinary/non-cloudinary props
-    const {cloudinaryProps, nonCloudinaryProps} = extractCloudinaryProps(
-      props, CloudinaryContext.CLOUDINARY_PROPS
-    );
-
-    //extract includeOwnBody which is used only by this context and should not be passed to children
-    const {includeOwnBody, ...childrenProps} = nonCloudinaryProps;
-
-    return {
-      context: cloudinaryProps,
-      childrenProps,
-      includeOwnBody
-    };
   }
 
   render() {
-    const {children} = this.props;
-    const {context, childrenProps, includeOwnBody} = this.state;
+    const {children, cloudinaryProps, nonCloudinaryProps, includeOwnBody} = splitProps(this.context, this.props);
```

Comment:
you started nicely with splitting all props to cloudinary/sdkSpecific/nonCloudinary, but didn't take it all the way...   Why not using cloudinaryReactProps.includeOwnBody in the ternary condition below?